frontend/public/components/operator-lifecycle-manager/descriptors/spec/index.tsx

```diff
@@ -30,18 +33,34 @@ const ResourceRequirements: React.SFC<SpecCapabilityProps> = ({obj, descriptor})
   <dd><ResourceRequirementsModalLink type="requests" obj={obj} path={descriptor.path} /></dd>
 </dl>;
 
-const K8sResourceLink: React.SFC<SpecCapabilityProps> = (props) => <ResourceLink kind={props.capability.split(SpecCapability.k8sResourcePrefix)[1]} name={props.value} namespace={props.namespace} title={props.value} />;
+const K8sResourceLink: React.SFC<SpecCapabilityProps> = (props) => _.isEmpty(props.value)
+  ? <span className="text-muted">None</span>
+  : <ResourceLink kind={props.capability.split(SpecCapability.k8sResourcePrefix)[1]} name={props.value} namespace={props.namespace} title={props.value} />;
 
 const BasicSelector: React.SFC<SpecCapabilityProps> = ({value, capability}) => <Selector selector={value} kind={capability.split(SpecCapability.selector)[1]} />;
 
+const BooleanSwitch: React.SFC<SpecCapabilityProps> = (props) => {
+  const patchFor = (value: boolean) => [{op: 'replace', path: `/spec/${props.descriptor.path.replace('.', '/')}`, value}];
+
+  return <div className="btn-group btn-group-sm" data-toggle="buttons">
+    <label className={classNames('btn compaction-btn', props.value ? 'btn-primary' : 'btn-default')}>
```

Comment:
Does this confirm the change before applying it? If not, I think we need to make this a two-click operation, either have a confirmation or have it so you click to show the toggle, then choose the option itself.

Reply:
Hmm, according to [this UX article](https://uxplanet.org/checkbox-vs-toggle-switch-7fc6e83f10b8), we should be using a checkbox instead if we don't want the action to be immediately applied. I still want inline confirm button, though...

Reply:
I am curious about you guys' thoughts if we could use the modal here? 
Something similar to this:
<img width="726" alt="tectonic-update-strategy" src="https://user-images.githubusercontent.com/5903705/47107864-463bf400-d1ff-11e8-85b3-7d517adf8f9b.png">

It's an existing "two-click operation" pattern in the console. We can expose "True" / "False" as radio buttons, and potentially add "help text" or default option if needed. What do you guys think?

Reply:
We haven't figured out how to incorporate modals with OLM descriptors yet. Plus I think a full-blown modal is overkill here, would prefer an inline solution.

Reply:
@tlwu2013 Added a confirmation step, see the screenshot in the PR description.

Reply:
@alecmerdler  Yeah, I see what are you saying. I think Toggle Switch should work since there's also a Toggle Switch in PatternFly.

FYI, @openshift/team-ux-review  
Per Alec, this PR is also using PF-react "switch" component, so should be good for the future migration to PF4.